### ISSUE TITLE
Basic user instructions?

### ISSUE THREAD

**Issue by snhirsch** [GryBsh/NabuNetworkEmulator#8]
Hi.  Sorry to be dense, but I cannot get nns to serve anything.  On my Ubuntu 18.04 system it comes up fine.  I connect the browser to port 5000 and am able to navigate to a screen that allows me to stop / start the USB serial serial server.  However, the NABU cannot boot and posts an "Adapter Not Authorized" message followed by Adapter Failure.  I'm not able to select different sources either.  There's a screen that appears to list different source packages, but there are no active hyperlinks.  I must be missing something basic, but not sure what.

Ooookay...  I figured out the issue.  This is something you may want to address, either in the code or the docs.  When I first connect the browser I see a 2-line screen at the top with an active link.  If I select the link, I'm taken to what *appears* to be the application, but the source select is not active - only the start/stop button.  The server is started, but not providing anything.  After a few rounds of this I turned away after the initial 2-line screen and it automatically went on by itself to the main screen.  But, now the link for program selection is active and able to serve properly.  Let me know if you need any more information?

**Comment by snhirsch** [comment]
Now that things are running, some additional feedback:  The first few times I started the server it responded very quickly to web connection.  After a while it became very sluggish and has remained that way through several restarts.  Interestingly, Cloud CP/M works much more reliably through this server than through the author's 'Internet Adapter'.  But, having said that, I cannot figure out where the CP/M Drive C: data is going!  I can write to that logical drive to my heart's content, but the cach/http:__cloud.nabu.ca_cpm_C.dsk file never gets updated and whatever I placed there is lost after shutdown.  

Update:  After I stop, delete the data.db file and restart performance is back to normal.

**Comment by snhirsch** [comment]
Another problem:  At startup I'm seeing several cascades of exceptions in this form:
`
info: [2023-06-08 11:57:24.2595] Nabu.Network.NabuNetwork[0]> Reading filesv2.txt from cache
fail: [2023-06-08 11:57:24.6990] Nabu.Network.NabuNetwork[0]> 
System.Net.Http.HttpRequestException: The SSL connection could not be established, see inner exception.
 ---> System.Security.Authentication.AuthenticationException: The remote certificate is invalid according to the validation procedure: RemoteCertificateNameMismatch
   at System.Net.Security.SslStream.SendAuthResetSignal(ProtocolToken message, ExceptionDispatchInfo exception)
   at System.Net.Security.SslStream.CompleteHandshake(SslAuthenticationOptions sslAuthenticationOptions)
   at System.Net.Security.SslStream.ForceAuthenticationAsync[TIOAdapter](Boolean receiveFirst, Byte[] reAuthenticationData, CancellationToken cancellationToken)
   at System.Net.Http.ConnectHelper.EstablishSslConnectionAsync(SslClientAuthenticationOptions sslOptions, HttpRequestMessage request, Boolean async, Stream stream, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Net.Http.ConnectHelper.EstablishSslConnectionAsync(SslClientAuthenticationOptions sslOptions, HttpRequestMessage request, Boolean async, Stream stream, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.ConnectAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.CreateHttp11ConnectionAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.AddHttp11ConnectionAsync(QueueItem queueItem)
   at System.Threading.Tasks.TaskCompletionSourceWithCancellation`1.WaitWithCancellationAsync(CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.HttpConnectionWaiter`1.WaitForConnectionAsync(Boolean async, CancellationToken requestCancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at Microsoft.Extensions.Http.Logging.LoggingHttpMessageHandler.<SendAsync>g__Core|5_0(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.Extensions.Http.Logging.LoggingScopeHttpMessageHandler.<SendAsync>g__Core|5_0(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)
   at Nabu.Network.CachingHttpClient.GetHead(String uri)
`

**Comment by GryBsh** [comment]
Apologies for the delay.
First, the above error is an issue downloading filesv2.txt from nabu.ca because of its security certificate, not much I can do about that I'm afraid. I cannot reproduce this, so I suspect it was a temporary issue of some kind. Unless it persists for you, then there is some issue validating DJs sites certificate on your machine. Let me know and we'll see what we can find out.

Second, yes, the loading screen does disappear on its own after 5 seconds by design. The text that appears is not a link, I suspect it was just appearing anyway. There is a link back to this github repo at the bottom at all times though.

I cannot reproduce a slow down over time from the database, but I have made several changes there in the upcoming release, so maybe I squashed that. I have a server running for sometimes weeks without any issue. 

You Cloud CPM woes are because nns doesn't persist changes to downloaded files. Remote changes will wipe out local ones because of how the cache works, it looks for changes and will download newer remote files right over the top of your changes. So, what I need to do is include a local C.dsk file in the storage source, and that will be in the pending release.

And when I say pending/upcoming, we're talking hours/tomorrow :)
